src/CompanyService.Validation/Users/CreateDepartmentUsersRequestValidator.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using FluentValidation;
+using LT.DigitalOffice.CompanyService.Models.Dto.Requests.User;
+using LT.DigitalOffice.CompanyService.Validation.Users.Interfaces;
+using LT.DigitalOffice.Kernel.Broker;
+using LT.DigitalOffice.Models.Broker.Common;
+using MassTransit;
+using Microsoft.Extensions.Logging;
+
+namespace LT.DigitalOffice.CompanyService.Validation.Users
+{
+  public class CreateDepartmentUsersRequestValidator : AbstractValidator<CreateDepartmentUsersRequest>, ICreateDepartmentUsersRequestValidator
+  {
+    private readonly IRequestClient<ICheckUsersExistence> _rcCheckUsersExistence;
+    private readonly ILogger<CreateDepartmentUsersRequestValidator> _logger;
+
+    public CreateDepartmentUsersRequestValidator(
+      IRequestClient<ICheckUsersExistence> rcCheckUsersExistence,
+      ILogger<CreateDepartmentUsersRequestValidator> logger)
+    {
+      _rcCheckUsersExistence = rcCheckUsersExistence;
+      _logger = logger;
+
+      RuleFor(x => x.UsersIds)
+        .Cascade(CascadeMode)
+        .NotEmpty().WithMessage("Users ids must not be null.")
+        .Must(x => x.Any()).WithMessage("Users ids must not be empty.")
+        .Must(x => !x.Contains(Guid.Empty)).WithMessage("Users ids must not contains empty value.")
+        .Must(x => x.Count() == x.Distinct().Count()).WithMessage("User cannot be added to the department twice.")
+        .Must(x => CheckUserExistence(x)).WithMessage("Users ids contains invalid id.");
+    }
+
+    private bool CheckUserExistence(List<Guid> usersIds)
+    {
+      try
+      {
+        var response = _rcCheckUsersExistence.GetResponse<IOperationResult<ICheckUsersExistence>>(
```

Comment:
result is bad, but how I understand we cannot use await here, so maybe we should move this to command?

Reply:
we can use await here, reworked